### PR TITLE
Feature/component bar

### PR DESCRIPTION
**Pull request by mrSelieznov**
#176  Task

Added component for settings bar and prepared for next step to implement it with service

### CODE REVIEW

File: src/components/SettingsBar.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { NavLink } from 'react-router-dom';
```
Comment (oleksandrstarov) [inline src/components/SettingsBar.tsx:1]: check indentation

File: src/App.scss [hunk 2]
```diff
@@ -9,6 +9,13 @@
     box-sizing: border-box;
 }
 
+hr{
```
Comment (oleksandrstarov) [inline src/App.scss:12]: add space

File: src/components/SettingsBtn.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { useState, useRef } from 'react';
```
Comment (oleksandrstarov) [inline src/components/SettingsBtn.tsx:1]: same as above

File: src/components/SettingsBtn.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    useOutsideClick(ref, () => {
+        if (bar) {
+            setBar(false);
+        }
+    });
+
+    const hendlerBar = () => {
+        setBar(!bar);
+    }
+    return (<div className="settings-wrapper">
```
Comment (oleksandrstarov) [inline src/components/SettingsBtn.tsx:22]: move jsx to new line

File: src/components/SettingsBtn.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { useState, useRef } from 'react';
+import useOutsideClick from 'src/hooks/useOutsideClick ';
+
+import '../styles/Header.scss';
+
+import settings from '../assets/settings.png';
+import SettingsBar from './SettingsBar';
+
+const SettingsBtn = () => {
+    const [bar, setBar] = useState(false);
```
Comment (oleksandrstarov) [inline src/components/SettingsBtn.tsx:10]: use appropriate name for boolean values, should start with is prefix

File: src/hooks/useOutsideClick .tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+
+import { useEffect } from "react";
```
Comment (oleksandrstarov) [inline src/hooks/useOutsideClick .tsx:2]: single quotes

File: src/hooks/useOutsideClick .tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+const useOutsideClick = (ref: any, callback: any) => {
```
Comment (oleksandrstarov) [inline src/hooks/useOutsideClick .tsx:4]: Please define types

File: src/hooks/useOutsideClick .tsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    const handleClick = (e: any) => {
+        if (ref.current && !ref.current.contains(e.target)) {
+            callback();
+        }
+    };
+
+    useEffect(() => {
+        document.addEventListener("click", handleClick);
```
Comment (oleksandrstarov) [inline src/hooks/useOutsideClick .tsx:12]: single quotes

File: src/styles/Header.scss [hunk 9]
```diff
@@ -87,4 +88,73 @@ header {
...
+      cursor: pointer;
+      height: 35px;
+      width: 35px;
+      border-radius: 50%;
+      background-color: #fff;
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      box-shadow: 0 0 3px rgba(0, 0, 0, 0.23);
+      // margin-right: 10px;
```
Comment (oleksandrstarov) [inline src/styles/Header.scss:105]: no commented code

File: src/styles/Header.scss [hunk 10]
```diff
@@ -87,4 +88,73 @@ header {
...
+  .settings-wrapper{
+    position: relative;
+
+    .settings-button{
+      position: relative;
+      cursor: pointer;
+      height: 35px;
+      width: 35px;
+      border-radius: 50%;
+      background-color: #fff;
```
Comment (oleksandrstarov) [inline src/styles/Header.scss:100]: should be variable I think

File: src/styles/SettingsBar.scss [hunk 11]
```diff
@@ -0,0 +1,80 @@
+@import "../styles/variables.scss";
+.link{
+  cursor:alias;
+}
```
Comment (oleksandrstarov) [inline src/styles/SettingsBar.scss:4]: separate by empty line

File: src/styles/Header.scss [hunk 12]
```diff
@@ -87,4 +88,73 @@ header {
       }
     }
   }
+  .settings-wrapper{
```
Comment (oleksandrstarov) [inline src/styles/Header.scss:91]: add empty line above

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/components/SettingsBar.tsx (hunk 1 above)
Comment (mrSelieznov) [reply src/components/SettingsBar.tsx:1]: if I understand correctly, I just added after the first line enter. So it fixed

File: src/components/SettingsBtn.tsx (hunk 3 above)
Comment (mrSelieznov) [reply src/components/SettingsBtn.tsx:1]: So it fixed

File: src/styles/SettingsBar.scss (hunk 11 above)
Comment (mrSelieznov) [reply src/styles/SettingsBar.scss:5]: image

File: src/styles/Header.scss (hunk 12 above)
Comment (mrSelieznov) [reply src/styles/Header.scss:94]: image

Comment (mrSelieznov) [reply src/styles/Header.scss:94]: fixed

**Review by Kachyr — APPROVED** [review]
check coments

File: src/components/SettingsBar.tsx (hunk 1 above)
Comment (oleksandrstarov) [reply src/components/SettingsBar.tsx:1]: Nope, all components in our app has 2 spaces line indent, this file has 4 - this was discussed a lot of times before, please set up your editor to use 2 spaces indentation

File: src/components/SettingsBar.tsx [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+                        <div className="bar-item-tool" title="Set to untracked history">
+                            <img src={focusImg} alt="focusImg" className="bar-img" />
+                            <div className="bar-clean-btn">
+                                <p className="bar-tool-title">Tracked</p>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+                <hr />
+
```
Comment (oleksandrstarov) [inline src/components/SettingsBar.tsx:61]: remove empty line

File: src/components/SettingsBar.tsx [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+            </div>
+        </div>
+
```
Comment (oleksandrstarov) [inline src/components/SettingsBar.tsx:64]: Remove empty line

File: src/components/SettingsBtn.tsx (hunk 3 above)
Comment (oleksandrstarov) [reply src/components/SettingsBtn.tsx:1]: it is not fixed, file contains indentation 4 spaces, should be 2

File: src/hooks/useOutsideClick .tsx [hunk 15]
```diff
@@ -0,0 +1,19 @@
+import { useEffect, RefObject } from 'react';
+
+const useOutsideClick = (ref: RefObject<HTMLElement>, callback: () => void) => {
+    const handleClick = (e: any) => {
```
Comment (oleksandrstarov) [inline src/hooks/useOutsideClick .tsx:4]: same

File: src/styles/Header.scss (hunk 12 above)
Comment (oleksandrstarov) [reply src/styles/Header.scss:94]: good, this file has 2 spaces as expected

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/components/SettingsBar.tsx (hunk 1 above)
Comment (mrSelieznov) [reply src/components/SettingsBar.tsx:1]: All the time I did it manually, but already I have changed it to automatically )

File: src/hooks/useOutsideClick .tsx [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick);
+
+    return () => {
+      document.removeEventListener('click', handleClick);
+    };
+  });
```
Comment (oleksandrstarov) [inline src/hooks/useOutsideClick .tsx:16]: shouldnt probably add listener to every render? add []

File: src/styles/Header.scss [hunk 17]
```diff
@@ -7,51 +9,51 @@ header {
   transition: all .4s ease-in-out;
   z-index: 100;
 
-  @media screen and(min-width: 993px) {
+  @media screen and(min-width: 993px)  {
```
Comment (oleksandrstarov) [inline src/styles/Header.scss:12]: 2 spaces here before { should be one

File: src/styles/SettingsBar.scss [hunk 18]
```diff
@@ -0,0 +1,77 @@
+@import "../styles/variables.scss";
+
+.settings-bar-wrapper {
+
+  .bar-items-container {
+
+  .bar-item {
```
Comment (oleksandrstarov) [inline src/styles/SettingsBar.scss:7]: add indent for child blocks

File: src/styles/SettingsBar.scss [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+      text-align: center;
+      line-height: 28px;
+      vertical-align: middle;
+
+      span {
+      font-size: 16px;
+      font-weight: bold;
+      color: white;
+      }
+    }
```
Comment (oleksandrstarov) [inline src/styles/SettingsBar.scss:47]: formatting

File: src/styles/SettingsBar.scss [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+      align-items:baseline;
+      cursor: pointer;
+
+      &:hover {
+      text-decoration: underline;
+      color: $purple;
+      }
+
+      .bar-img {
+      height: 20px;
```
Comment (oleksandrstarov) [inline src/styles/SettingsBar.scss:64]: formatting

File: src/components/SettingsBtn.tsx [hunk 21]
```diff
@@ -0,0 +1,34 @@
+import { useState, useRef } from 'react';
+
+import useOutsideClick from 'src/hooks/useOutsideClick ';
+import '../styles/Header.scss';
+
+import settings from '../assets/settings.png';
+import SettingsBar from './SettingsBar';
+
+const SettingsBtn = () => {
+    const [isBar, setIsBar] = useState(false);
```
Comment (oleksandrstarov) [inline src/components/SettingsBtn.tsx:10]: not changed here

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/hooks/useOutsideClick .tsx (hunk 16 above)
Comment (mrSelieznov) [reply src/hooks/useOutsideClick .tsx:16]: I guess shouldn't add listener It works precisely

File: src/styles/Header.scss [hunk 22]
```diff
@@ -87,4 +90,73 @@ header {
...
+      padding: 8px ;
+      position: absolute;
+      height: 257px;
+      width: 235px;
+      opacity: 1;
+      z-index: 10;
+      display: block;
+    }
+  }
+
```
Comment (oleksandrstarov) [inline src/styles/Header.scss:161]: Please remove empty line

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/components/SettingsBtn.tsx [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+  const hendlerBar = () => {
+    setIsBar(!isBar);
+  }
+  return (
+    <div className="settings-wrapper">
+      <div onClick={() => hendlerBar()} className="header-nav-link settings-button">
+        <img src={settings} alt="settings img" className={!isBar ? "settings-img-button" : "settings-img-button-active"} />
+      </div>
+      <div ref={ref} className={!isBar ? "settings-bar" : "settings-bar-active"}>
+        <SettingsBar />
```
[the remark this answers is not in the document]
Comment (mrSelieznov) [reply src/components/SettingsBtn.tsx:28]: if isSettingsBarActive equal true or false, it should be rendered anyway but like you can see with referent classes

File: src/styles/SettingsBar.scss [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+            width: 20px;
+            margin-right: 13px;
+          }
+          
+          .bar-clean-btn {
+      
+            .bar-tool-title {
+              
+            }
+          }
```
Comment (oleksandrstarov) [inline src/styles/SettingsBar.scss:74]: ???

File: src/components/SettingsBtn.tsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (oleksandrstarov) [reply src/components/SettingsBtn.tsx:29]: settings-bar class has display-none - so I guess it wont be rendered for user. Any reason to keep it then?

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

**Review by oleksandrstarov — APPROVED** [review]

**Pull request merged**